Makefile.am: correct path to libtool

wolfsage opened the pull request:
This fix comes from https://github.com/cyrusimap/cyrus-imapd/pull/5186/

mjdominus committed:
Makefile.am: correct path to libtool

elliefm commented on Makefile.am:
```diff
@@ -698,7 +698,8 @@ cunit/registers.h: $(CUNIT_PROJECT)
 
 # To run under Valgrind, do: make VG=1 check
-VALGRIND = libtool --mode=execute valgrind \
+# The path to libtool is relative to the cunit/ directory
```
This comment is weird?  It might be explaining the `${abs_top_srcdir}` variable, but I don't think that's the detail that needs explaining here, it's just an ordinary part of automake.

IMO the interesting (and surprising to me) detail is that we're now using a libtool from the source directory, rather than the system libtool from `$PATH` (in my case, `which libtool` says `/usr/bin/libtool`).  And we do indeed have a libtool in the source directory!  Looks like it gets generated during configure.  Knowing that it's there, using it smells correct, but I'd never thought about this before.  On the command line, I've always used the system one (e.g. for invoking gdb/ldd/etc on some executable, like `libtool --mode=execute gdb imap/imapd` or whatever), so I should retrain my fingers.

I think we mostly don't invoke libtool directly during build, but rather the stuff generated by autotools does it correctly for us when it has to, and this case is exceptional because these rules were hand-written.  So I do wonder if there's other hand-written invocations of it like this that have been missed....  Scanning the output of `ack libtool`, I don't see any other misuses of it, but there is one spot in our documentation that needs updating too:

```
docsrc/imap/reference/faqs/o-gdb.rst
24:source/build tree), you need to run gdb from ``libtool`` in order for everything
31:    $ libtool --mode=execute gdb imap/cyr_virusscan
```

I think that invocation on line 31 should be `./libtool`.  Can you fix that while you're in here?  Thanks

And given that I now think it's correct to use the source directory libtool and not the system one, I guess the comment that originally bothered me doesn't need to exist at all.  We were using the wrong libtool, now we're using the right libtool, and the commit message adequately explains that.

wolfsage replied:
I've removed that comment and found a few more references. See what you think? Thanks

elliefm reviewed (changes requested):
Fascinating! Detailed commentary inline

wolfsage committed:
docs: Update other libtool references to use local built copy

We don't install libtool-bin, instead we expect a libtool script to exist in
the base directory, so we should use that instead.

elliefm commented:
**Backporting:** probably grab #4566 too
